common/src/stack/report-system/command/report/system/fixtures/disk_partition_check.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import paramiko
+import socket
+import re
+from collections import namedtuple
+from itertools import groupby
+
+# Takes in a hostname, returns the lsblk data in a dictionary of disks with a nested
+# dictionary of partitions per disk which each has a named tuple of the name, mountpoint,
+# disk label, file system type, and disk size. Returns an empty dictionary if the host
+# cannot be ssh into or lsblk does not return any data
+def get_partition_data(hostname):
+
+	disks = {}
+	partitions = {}
+
+	backend_ssh = paramiko.SSHClient()
+	backend_ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+	# Skip the test if we can't ssh in
+	try:
+			backend_ssh.connect(hostname=f'{hostname}')
+
+	except (paramiko.SSHException, socket.error):
+		return {}
+
+	# Get backend host partition informaiton using lsblk
+	stdin, stdout, stderr = backend_ssh.exec_command('lsblk -r -n -o name,mountpoint,label,fstype,size')
+
+	lsblk_data = stdout.readlines()
+
+	# Turn each key value string pair into it's own entry into the list
+	if lsblk_data:
+		host_part_data = ' '.join(lsblk_data).replace('\n', '').split(' ')
```

Comment:
The code from line 34 => 51 seem like a really roundabout way to parse the lines your are getting back. If each partition is coming out of lsblk on a single line, you should be able to just split and insert the data without needing the zip stage:

```
Partition = namedtuple("Partition", ['name', 'mountpoint', 'label', 'fstype', 'size'])
for line in lsblk_data:
    part_tuple = Partition(*line.split())
```